sphinx_click/ext.py

```diff
@@ -79,12 +79,17 @@ def _write_opts(opts: ty.List[str]) -> str:
 
     extras = []
 
-    if isinstance(opt.show_default, str):
+    if opt.show_default is not None:
+        show_default = opt.show_default
+    else:
+        show_default =  ctx.show_default
```

Comment:
This is available [in click 7.x](https://click.palletsprojects.com/en/7.x/api/?highlight=context#click.Context) so we don't need version-specific logic :+1:

However, the [click 8.1.x docs state](https://click.palletsprojects.com/en/8.1.x/api/?highlight=context#click.Context):

> Changed in version 8.1: The show_default parameter is overridden by Command.show_default, instead of the other way around.

Should we replicate this behaviour here? It feels like we should? If we don't, we might want to add a note here that we intentionally decided not to.